src/metadata.jl

```diff
@@ -20,12 +20,17 @@ const _CACHE = Dict(
     :IMAGE => "",
 )
 
+# Make sure _CACHE access is thread-safe
+const _CACHE_LOCK = ReentrantLock()
+
 function _pkgs()
-    if isempty(_CACHE[:PKGS])
-        for (pkg, ver) in Pkg.installed()
-            # BSON can't handle Void types
-            if ver !== nothing
-                global _CACHE[:PKGS][pkg] = ver
+    lock(_CACHE_LOCK) do
+        if isempty(_CACHE[:PKGS])
+            for (pkg, ver) in Pkg.installed()
```

Comment:
Hmm, I think we should have a failing test cases that's fixed by this change? I wonder if we should mock the `Pkg.installed()` call in order to write an initially failing test for the non-threadsafe version? @oxinabox Do you have any opinions/suggestions on this?

Reply:
I agree a test would nice. But I'm not sure how to write one in this case.

Reply:
I can't think how to do this reliably.
Because the scheduler never promises to run things concurrently